src/scales/scale.ts

```diff
@@ -122,19 +118,23 @@ module Plottable {
       }
     }
 
-    protected _getDomain() {
-      return this._d3Scale.domain();
+    protected _getDomain(): D[] {
+      throw new Error("subclasses should override _getDomain");
     }
 
     protected _setDomain(values: D[]) {
       if (!this._domainModificationInProgress) {
         this._domainModificationInProgress = true;
-        this._d3Scale.domain(values);
+        this._setActualDomain(values);
         this._dispatchUpdate();
         this._domainModificationInProgress = false;
       }
     }
 
+    protected _setActualDomain(values: D[]) {
```

Comment:
What's this? This name is really confusing, since there's already a `_setDomain()` method.